pkg/chartutil/load.go

```diff
@@ -94,8 +95,14 @@ func LoadArchive(in io.Reader) (*chart.Chart, error) {
 			delimiter = "\\"
 		}
 
+		if !strings.Contains(hd.Name, delimiter+"templates"+delimiter) && !strings.Contains(hd.Name, delimiter+"charts"+delimiter) {
+			if strings.Count(hd.Name, delimiter) < chartBeginIdx {
+				chartBeginIdx = strings.Count(hd.Name, delimiter)
+			}
+		}
+
```

Comment:
This code is in place to handle the case of a chart tar containing `nestedDir/chartName/<chart contents>` -- without it helm doesn't understand the generated tar. Granted, this is only generated this way because of the below-mentioned `name: nestedDir/fooChart` metadata values.